Tools/clinic/clinic.py

```diff
@@ -2776,7 +2776,7 @@ def _render_non_self(self, parameter, data):
         if cleanup:
             data.cleanup.append('/* Cleanup for ' + name + ' */\n' + cleanup.rstrip() + "\n")
 
-    def render(self, parameter, data):
+    def render(self, parameter: str, data: CRenderData) -> None:
```

Comment:
This looks fishy. `parameter` should be a `Parameter` instance, but also `str` is accepted. Let's look into that in a follow-up PR.